Source/WebCore/rendering/RenderLayer.h

```diff
@@ -969,10 +963,24 @@ class RenderLayer : public CanMakeWeakPtr<RenderLayer>, public CanMakeCheckedPtr
     void setAncestorChainHasSelfPaintingLayerDescendant();
     void dirtyAncestorChainHasSelfPaintingLayerDescendantStatus();
 
+    struct RepaintRects {
+        LayoutRect clippedOverflowRect;
+        LayoutRect outlineBoundsRect;
+    };
+
+
+    std::optional<RepaintRects> repaintRects() const
+    {
+        if (m_repaintRectsValid)
```

Comment:
nit, not super consistent with RenderLayer::cachedClippedOverflowRect() which you did the opposite and used std::nullopt.